### PR TITLE
Use Towncrier to create Changelog

### LINKED ISSUE #193: Missing release notes
I couldn't find a CHANGES.md, or section in the README, or release notes added via the GitHub 'releases' feature.

I don't suppose some could be added for future releases?

### PR DESCRIPTION
**Pull request by djmitche**
Fixes #193.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Making a Release
+
+* Run `towncrier --version=$newversion --draft` and check that the output looks OK.  Then run it without `--draft`, deleting the old newsfiles.  Commit.
+* Update the version, commit, and tag -- `npm version patch` (or minor or major, depending)
+* Push to release the JS version -- `git push && git push --tags`
+* Release to PyPi:
+  * `python setup.py sdist`
+  * `twine upload dist/json-e-<version>.tar.gz`
+* Go doesn't believe in versions, so there's notihng to do for a Go release!
```
Comment (jonasfj) [inline .github/CONTRIBUTING.md]: For golang we could use http://gopkg.in` and then tags in github...

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+of `.bugfix`, `.feature`, or (for docs-only changes) `.doc`.  The content of
+this file should be in reStructuredText format. Keep it to a simple sentence,
+and you should be fine!
+
+For example, `newsfragments/201.bugfix` might contain `Fixed the precedence of
+the || and 'in' operators`.
+
+# Making a Release
+
+* Run `towncrier --version=$newversion --draft` and check that the output looks OK.  Then run it without `--draft`, deleting the old newsfiles.  Commit.
```
Comment (jonasfj) [inline .github/CONTRIBUTING.md]: This is too complicated, make it a script and list the credentials that you must have in-order to make a release...

Or even better, let's make travis responsible for releases and just make releasing the result of pushing a tag...
Or running a script that does `towncrier` and pushes a tag...

File: jsone/newsfragments/193.feature [hunk 3]
```diff
@@ -0,0 +1 @@
+Begin using towncrier to build CHANGELOG.rst
```
Comment (jonasfj) [inline jsone/newsfragments/193.feature]: I'm not a fan... I would much prefer some test logic that enforced sane commit messages... And then we build changelog from commit messages when we make a new release...

Also why is it in the `jsone` sub-folder? I'm not sure where it should be? and should it be called `newsfragments/` why not `.changelog/`.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+start`) there.
+
+The resulting application embeds and enriches the README file.
+
+# Changelog
+
+When making a pull request with your changes, create a new file in
+`newsfragments/` named after the issue or bug you are working on, with a suffix
+of `.bugfix`, `.feature`, or (for docs-only changes) `.doc`.  The content of
+this file should be in reStructuredText format. Keep it to a simple sentence,
```
Comment (jonasfj) [inline .github/CONTRIBUTING.md]: Why not markdown?

Couldn't we just have a script pull it from github commit log... like: https://github.com/conventional-changelog/standard-version

**Comment by djmitche** [comment]
@edmorley made some good arguments against considering the git log to be a changelog, and the towncrier README does, too:

> towncrier delivers the news which is convenient to those that hear it, not those that write it.
> 
> That is, by duplicating what has changed from the “developer log” (which may contain complex information about the original issue, how it was fixed, who authored the fix, and who reviewed the fix) into a “news fragment” (a small file containing just enough information to be useful to end users), towncrier can produce a digest of the changes which is valuable to those who may wish to use the software. These fragments are also commonly called “topfiles” or “newsfiles” in Twisted parlance.

The limitations here (reStructuredText, calling it newsfragments, putting it in the python package directory) are just the way towncrier works.  My intent here was to do the simplest thing that would make @edmorley happy, not to embark on a project :)

@jonasfj it sounds like you're volunteering to set this up with another tool, with some on-tag automation, maybe some fanciness to create new release objects in github.  Can I assign #193 to you?

**Comment by jonasfj** [comment]
> My intent here was to do the simplest thing that would make @edmorley happy, not to embark on a project :)

Fair point, I might be over thinking this...

**Comment by jonasfj** [comment]
So IMO, we should still script the release process... but that can be a separate issue...

**Pull request merged**